platform/features/conductor-redux/src/README.md

```diff
@@ -0,0 +1,187 @@
+//Somewhere in the TimeConductorController...
+function changeTOI(newTime) {
+    MCT.conductor.timeOfInterest(newTime);
+}
+```
+### 3. Plot controller listens for change to TOI
+```javascript
+// toi is attribute of Time Conductor object. Add a listener to the time 
+// conductor to be alerted to changes in value
+
+MCT.conductor.listen(function (conductor) {
+    var toi = conductor.timeOfInterest();
+    plot.setTimeOfInterest(toi);
+});
+```
+### 4. Real-time object is loaded and time conductor is set up in real-time mode showing last 15 minutes of data.
+```javascript
+// Notional initialization function, perhaps in a 'runs' extension
+function initializeConductor(telemetryMetadata) {
+    var tc = MCT.conductor,
+        FIFTEEN_MINUTES = 15 * 60 * 1000;
+    tc.timeSystem(telemetryMetadata.timeSystem);
+    
+    //Start and end modes set separately (to support fixed start, follow 
+    // time end. Follow time = RELATIVE end mode where delta value = 0) 
+    tc.startMode(tc.Modes.RELATIVE);
+    tc.endMode(tc.Modes.RELATIVE);
+    
+    // Delta defined in ms. Start time delta is relative to end time.
+    tc.bounds.inner.start.delta(FIFTEEN_MINUTES);
+    tc.bounds.outer.start.delta(FIFTEEN_MINUTES);
+    
+    // Delta defined in ms. An end time delta value is relative to 'now'. 
+    // Thus, an end delta of zero is 'follow time'. May be > 0 to track 
+    // time in the future.
+    // Could add a convenience function to TC to make setting follow 
+    // time mode a little clearer
+    tc.bounds.inner.end.delta(0);
+    tc.bounds.outer.end.delta(0);
+}
+```
+### 5. Switch to follow time. Set inner end to now, and outer end to now + 1 hour
+
+```javascript
+
+var tc = MCT.conductor;
+
+// Somewhere in the time conductor controller...
+function setFollowTime() {
+    var ONE_HOUR = 1 * 60 * 60 * 1000;
+    
+    tc.endMode(tc.bounds.Modes.RELATIVE);
+    tc.startMode(tc.bounds.Modes.RELATIVE);
+    
+    //Set start deltas. Start deltas always relative to end
+    tc.bounds.inner.start.delta(-ONE_HOUR);
+    tc.bounds.outer.start.delta(-ONE_HOUR);
+    
+    //Set end deltas. End deltas always relative to `now`
+    tc.bounds.inner.end.delta(0);
+    tc.bounds.outer.end.delta(ONE_HOUR);
+}
+```
+
+### 6. Plot Controller updated on tick
+``` javascript
+// Plot only really interested in inner bounds.
+// Specify event type of TICK to receive tick events only.
+MCT.conductor.bounds.inner.listen(function (innerBounds) {
+    plotUpdater.setDomainBounds(bounds.start.time(), bounds.end.time());
+}, MCT.conductor.bounds.EventTypes.TICK);
+```
+
+### 7. Plot Controller updated when user changes bounds
+``` javascript
+// Plot only really interested in inner bounds
+// Specify event type of USER to receive user events (such as bounds 
+// handles being dragged) only.
+MCT.conductor.bounds.inner.listen(function(bounds) {
+    plotUpdater.setDomainBounds(bounds.start.time(), bounds.end.time());
+}, MCT.conductor.bounds.EventTypes.USER);
```

Comment:
@VWoeltjen Good points. Yes, from the outset I made the assumption that we would be using our current observable pattern approach, without really considering alternatives. You're right, this seems like a good time to standardize our approach here for the broader API.

I did have an earlier iteration that was quasi-EventEmitter like but with enum matching instead of string matching. I convinced myself that it was simpler and clearer to just attach listeners to the object you're interested in (eg. the inner bounds) than to specify strings or enum values for the event type you're interested in, but I'm not so sure about it and I keep wavering. Having a single listener attachment point on the TimeConductor does make a lot of sense, and removes any ambiguity about which object to listen to. Being able to properly document this with JSDoc will increase clarity about event names.

Can see advantages to event streams for continuous events like ticks.

Tend to agree with your assessment. EventEmitter sounds like a good approach.